src/iarc7_motion/iarc_tasks/block_roomba_task.py

```diff
@@ -0,0 +1,231 @@
+                    drone_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        'map',
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('Block Roomba Task: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during block roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not close enough to the quad'),)
+                if self._on_ground():
+                    return (TaskDone(),)
+
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+                roomba_velocity = math.sqrt(roomba_x_velocity**2 + roomba_y_velocity**2)
+
+                roomba_vector = Vector3Stamped()
+
+                roomba_vector.vector.x = roomba_x_velocity/roomba_velocity
+                roomba_vector.vector.y = roomba_y_velocity/roomba_velocity
+                roomba_vector.vector.z = 0.0
+
+                # p-controller
+                x_vel_target = ((self._roomba_point.point.x + self._overshoot * roomba_vector.vector.x)
+                                    * self._K_X + roomba_x_velocity)
+                y_vel_target = ((self._roomba_point.point.y + self._overshoot * roomba_vector.vector.y)
+                                    * self._K_Y + roomba_y_velocity)
+                
+                z_vel_target = self._descent_velocity
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_TRANSLATION_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+                
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+                desired_vel = []
```

Comment:
Make into single array? 4 lines isn't needed you can do desired_vel = [x_vel_target, y_vel_target,...]